src/components/Contexts/ErrorMessageContext.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext, useState } from 'react';
+
+interface ErrorContextType {
+  errorMessage: string,
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>
+  isErrorHidden: boolean,
+  setIsErrorHidden: React.Dispatch<React.SetStateAction<boolean>>,
+  handleShowError: (error: string) => void,
+}
+
+export const ErrorMessageContext = React.createContext({} as ErrorContextType);
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const ErrorMessageContextProvider: React.FC<Props> = ({ children }) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isErrorHidden, setIsErrorHidden] = useState(true);
```

Comment:
I want my error to disappear smoothly, if I just setErrorMessage('') it will remove text and after hides the block. Which seems incorrect for me.
https://drive.google.com/file/d/1murkaSWOv1Q86OYh3rrXKQ-SDNyChk88/view?usp=sharing